test/hack/bin/check-opsrecipes.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+set -euo pipefail
+
+# List of generated rules
+RULES_FILES=(./test/hack/output/*/prometheus-rules/templates/alerting-rules/*)
+#RULES_FILES=(./test/hack/output/*/prometheus-rules/templates/alerting-rules/up*)
+
+DEBUG_MODE=false
+CHECK_EXTRADATA_ERRORS=false
+CHECK_NORECIPE_ERRORS=true
+CHECK_UNEXISTINGRECIPE_ERRORS=true
+
+# Parameters:
+# - an element
+# - an array
+# Returns:
+# - 0 if element is not found in the array
+# - 1 if element is found in the array
+isInArray () {
+    referenceElement="$1" && shift
+    array=("$@")
+
+    for element in "${array[@]}"; do
+        [[ "$element" == "$referenceElement" ]] \
+            && return 0
+    done
+    return 1
+}
+
+listOpsRecipes () {
+    # find list of opsrecipes from git repo
+    tmpDir="$(mktemp -d)"
+    git clone --depth 1 --single-branch -b main -q git@github.com:giantswarm/giantswarm.git "$tmpDir"
+    find "$tmpDir"/content/docs/support-and-ops/ops-recipes -name \*.md \
+        | sed -n 's_'"$tmpDir"'/content/docs/support-and-ops/ops-recipes/\(.*\).md_\1_p'
+    rm -rf "$tmpDir"
+
+    # Add extra opsrecipes
+    # These ones are defined as aliases of `deployment-not-satisfied`:
+    echo "workload-cluster-managed-deployment-not-satisfied"
+    echo "workload-cluster-deployment-not-satisfied"
+    echo "deployment-not-satisfied-china"
```

Comment:
Is there a better way maybe to handle alias than hardcoding values here ?

Reply:
I considered it, and it seemed like a lot of unsafe code for not much.
I confess I did not try to code it, just tried to imagine it and didn't feel inspired.

Reply:
Ok fine then.